src/board.rs

```diff
@@ -243,10 +246,8 @@ impl Board {
     pub fn get_piece_at_position(
         &self,
         square: Square,
-    ) -> Result<Option<(PieceType, Side)>, InvalidSquareError> {
-        self.validate_square(square).map(|square| {
-            self.squares[square.rank * self.width + square.file]
-        }) 
+    ) -> Option<(PieceType, Side)> { 
```

Comment:
I do not think this signature should be changed. We should keep validating squares inside, instead. The value of this function is that it will always return something that's a valid square, and so consumers can rely on that.

Also, I think you meant to commit these changes earlier, builds fail with earlier commits because the signatures aren't updated. This is also part of why I think we should keep the same signatures, and just modify the internals

Reply:
Hmm... on second look, is this commit specifically just relying on the Piece just... existing, since Square types are always valid?

Reply:
Yeah, the function will always have a valid square, the Option there is whether there's a piece there or not.